test/unit/Reflection/ReflectionClassTest.php

```diff
@@ -425,6 +426,93 @@ public function testGetParentClassWithMissingParent(): void
         $classInfo->getParentClass();
     }
 
+    public function testGetInterfaceNamesWithMissingInterfaceDefinitions(): void
+    {
+        $classInfo = (new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+            $this->astLocator,
+        )))->reflectClass(ClassWithMissingInterface::class);
+
+        $this->expectException(IdentifierNotFound::class);
+
+        self::assertNotNull($classInfo->getInterfaceNames());
+    }
+
+    public function testGetInterfacesWithMissingInterfaceDefinitions(): void
+    {
+        $classInfo = (new DefaultReflector(new SingleFileSourceLocator(
+            __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+            $this->astLocator,
+        )))->reflectClass(ClassWithMissingInterface::class);
+
+        $this->expectException(IdentifierNotFound::class);
+
+        self::assertNotNull($classInfo->getInterfaces());
+    }
+
+    /** @param list<class-string> $expectedInterfaces */
+    #[DataProvider('getInterfaceClassNamesDataProvider')]
+    public function testGetInterfaceClassNames(string $sourcePath, string $className, array $expectedInterfaces): void
+    {
+        $classInfo = (new DefaultReflector(new SingleFileSourceLocator(
+            $sourcePath,
+            $this->astLocator,
+        )))->reflectClass($className);
+
+        self::assertSame(
+            $expectedInterfaces,
+            $classInfo->getInterfaceClassNames(),
+        );
+    }
+
+    /** @return list<array{string, class-string, list<class-string>}> */
+    public static function getInterfaceClassNamesDataProvider(): array
+    {
+        return [
+            [
+                __DIR__ . '/../Fixture/ClassWithMissingInterface.php',
+                ClassWithMissingInterface::class,
+                ['Roave\BetterReflectionTest\Fixture\InterfaceThatDoesNotExist'],
+            ],
```

Comment:
this first test-case shows that we are now able to extract the interface names even if not autoloadable.

the remaining cases add coverage a few more related cases like autoloadable classes, traits,..